src/style-spec/validate/validate.js

```diff
@@ -62,7 +62,18 @@ export default function validate(options) {
         return validateExpression(options);
 
     } else if (valueSpec.type && VALIDATORS[valueSpec.type]) {
-        return VALIDATORS[valueSpec.type](options);
+        const errors = VALIDATORS[valueSpec.type](options);
```

Comment:
Can this be simplified, if I understand we need to concatenate errors with legacy errors if there is legacy type, right?

Reply:
yeah, this is what's happening below. We first check the spec type, if fail we check legacy type, if fail again - concatenate the errors.